app/helpers/vm_helper/textual_summary.rb

```diff
@@ -257,35 +257,15 @@ def textual_group
   end
 
   def textual_ems
-    ems = @record.ext_management_system
-    return nil if ems.nil?
-    label = ui_lookup(:table => "ems_infra")
-    h = {:label => label, :image => "vendor-#{ems.image_name}", :value => ems.name}
-    if role_allows(:feature => "ems_infra_show")
-      h[:title] = "Show parent #{label} '#{ems.name}'"
-      h[:link]  = url_for(:controller => 'ems_infra', :action => 'show', :id => ems)
-    end
-    h
+    textual_link(@record.ext_management_system, :as => EmsInfra)
   end
 
   def textual_cluster
-    cluster = @record.ems_cluster
-    h = {:label => title_for_cluster, :image => "ems_cluster", :value => (cluster.nil? ? "None" : cluster.name)}
-    if cluster && role_allows(:feature => "ems_cluster_show")
-      h[:title] = "Show this VM's #{title_for_cluster}"
-      h[:link]  = url_for(:controller => 'ems_cluster', :action => 'show', :id => cluster)
-    end
-    h
+    textual_link(@record.ems_cluster)
```

Comment:
Oops... this was exactly the most frequent/minor variation that I was supposed to be skipping (per "deliberately conservative").